src/main/java/nwoolcan/utils/Result.java

```diff
@@ -162,6 +162,32 @@ public T orElse(final Supplier<? extends T> other) {
     public Result<T> require(final Supplier<Boolean> supplier) {
         return this.require(supplier, new IllegalArgumentException());
     }
+    /**
+     * Apply the provided function to the exception, if any and the exception type matches
+     * the provided one.
+     * @param exception the type of exception.
+     * @param function the function to apply to the exception
+     * @param <E> the type of the exception
+     * @return this
+     */
+    @SuppressWarnings("unchecked")
+    public <E extends Exception> Result<T> peekError(final Class<E> exception, final Consumer<E> function) {
+        if (this.isError() && exception.equals(this.exception.get().getClass())) {
+           function.accept((E) this.exception.get());
+        }
+        return this;
+    }
+    /**
+     * Apply the provided function to the exception, if any.
+     * @param function the function to apply to the exception
+     * @return this
+     */
+    public Result<T> peekError(final Consumer<Exception> function) {
```

Comment:
Shouldn't it be:
`public Result<T> peekError(final Consumer<? extends Exception> function)` ?

Reply:
No because that method will always be applied if the current `Result` holds an exception. The notation `public Result<T> peekError(final Consumer<? extends Exception> function)` could lead to invoking the function with, for instance, a `Consumer<NullPointerException>`, while inside the `Result` has a `IllegalStateException`. Other than that, it won't compile without changing the underlying `Result` because of the reasons discussed above.

Reply:
Yes, you are right.
I think maybe can be use with `Consumer<? super Exception>` instead maybe?
Still its fine ad it is.